src/api/time/TimeAPI.js

```diff
@@ -152,6 +153,12 @@ class TimeAPI extends GlobalTimeContext {
       timeContext.setMode(FIXED_MODE_KEY, value);
     }
 
+    // Also emit the mode in case it's different from the previous time context
+    timeContext.emit(
+      TIME_CONTEXT_EVENTS.modeChanged,
+      JSON.parse(JSON.stringify(timeContext.getMode()))
```

Comment:
I think we can use `structuredClone` over `JSON` cloning, unless there is a good reason not to.